### ISSUE TITLE
Empty list on feed

### ISSUE THREAD

**Issue by pascalberger** [chocolatey/ChocolateyGUI#503]
I get an empty list on the remote feed on page 3:

* Open Chocolatey GUI
* Switch to source (bbt in this case)
* Packages are shown. Click next to go to second page
* Packages are shown. Click next to go to third page
* Empty page is displayed

![image](https://user-images.githubusercontent.com/2190718/31810980-19414584-b57e-11e7-882e-36a03e0aa2a5.png)

**Comment by gep13** [comment]
@pascalberger out of interest, how many packages do you have on that feed?  i.e. "should" there be anything on that third page?

**Comment by gep13** [comment]
@pascalberger if/when you have a minute, can you test this out again with the latest from the MyGet Feed?  The new version now includes a page number at the bottom of the page.  I am curious if this shows anything that could help.

**Comment by pascalberger** [comment]
@gep13 Tried with v0.16.0-unstable0046. Going to Community Feed source, click next empty page appears:

![grafik](https://user-images.githubusercontent.com/2190718/32131629-912362e0-bbb3-11e7-919d-033315ebe58e.png)

**Comment by gep13** [comment]
I wonder if you are seeing some of the caching issues that happen from time to time, i.e. why are you only getting 6 packages in the list.  /cc @ferventcoder

**Comment by RichiCoder1** [comment]
^ This. I've been seeing some weird issues myself.

**Comment by ferventcoder** [comment]
No idea here myself. Could be issues with the remote repo. Might be a good idea to check there and see if it experiences the same.

**Comment by pascalberger** [comment]
I also had this on our internal ProGet hosted feed. So definitely not only a community feed caching issue

**Comment by gep13** [comment]
@pascalberger is it happening reliably, or is it intermittent?

**Comment by pascalberger** [comment]
@gep13 Reliable. ProGet shows 110 packages on the feed.

**Comment by gep13** [comment]
@pascalberger ok, will likely need to you provide some logs, and perhaps some output from running Chocolatey directly as well.  @RichiCoder1 any thoughts on exactly what we would need in order to figure out what the issue is here?

**Comment by gep13** [comment]
@pascalberger I have been able to replicate at least part of this problem by using the chocolatey.org feed.

If I navigate to the last page there (which is 131) I get an empty page.  If I go to the page before, I get 46 results returned.  So something is off somewhere.

Internally, Chocolatey GUI is using a method in Chocolatey.Lib to return the total package count (which is currently coming back as 6527) which tallies with the packages currently on chocolatey.org:

![image](https://user-images.githubusercontent.com/1271146/56741843-8a3e2180-676b-11e9-914c-1bc1f899767e.png)

However, based on paging through the results, I am getting the total number of packages to be 6496, not 6527.

i.e. (129 * 50) + 46 = 6496

If I run the command:

`choco list --source https://chocolatey.org/api/v2`

and look at the number returned at the end, I get a different number again:

![image](https://user-images.githubusercontent.com/1271146/56741996-db4e1580-676b-11e9-8e13-2a2b6d1e89af.png)

While this number is "closer" the the number returned when paging through the results, it still isn't the same (off by 1).

Long story short...

I don't think that this is an issue with Chocolatey GUI, but not sure yet on how to prove, or correct that.

**Comment by pascalberger** [comment]
My issue was with a ProGet hosted feed. So, issue seems also not related to server implementation (assuming we're talking about a single issue only here). This would only leave Chocolatey.Lib left, responsible for this.

**Comment by gep13** [comment]
@pascalberger as yet, I have been unable to replicate this issue.  As such, I am going to remove this from the 0.18.0 milestone.  And will continue to monitor for this issue happening, and to try to get a reproducible test that would allow us to figure out what is going on with it.
